### ISSUE TITLE
notation for tpair?

### ISSUE THREAD

**Issue by DanGrayson** [UniMath/UniMath#181]
I tried making a "recursive" notation for tpair, and the best I could do is this:
```
Notation "( x ; .. ; y , z )" := (tpair _ x .. (tpair _ y z) ..) : core_scope.
(* Examples: *)
Goal Σ (_:nat), nat. exact (4,5). Defined.
Goal Σ (_:nat) (_:nat) (_:nat) (_:nat), nat. exact (2;3;4;5,6). Defined.
```
That's due to some peculiarity in the parser, which gets confused if you try to make all the separators into commas.  Also, there can be an annoying warning message from coqtop about masking a rule in the parsing grammar:
```
<W> Grammar extension: in [constr:operconstr] some rule has been masked
<W> Grammar extension: in [constr:pattern] some rule has been masked
```
Changing the comma to something else would eliminate the warning message.

Comments?

**Comment by DanGrayson** [comment]
Another possibility is to not insist on parentheses, as in : 
```
Notation "x , y" := (tpair _ x y) (at level 69, right associativity).
```

**Comment by DanGrayson** [comment]
Resolved in commit 05cfc1d92543f5e10ecfbfa5067d2b25e44eee4e
